src/dgrid/DgridWrapper.ts

```diff
@@ -1,17 +1,637 @@
+	grid: DgridGrid;
+	columnId: string | number;
+	width: number;
+	parentType?: string;
+}
+
+interface DgridSortEvent extends Event {
+	grid: DgridGrid;
+	parentType?: string;
+	sort?: { property: string; descending?: boolean }[];
+}
+
+interface DgridErrorEvent extends Event {
+	grid: DgridGrid;
+	error?: Error | string;
+}
+
+interface DgridCellFocusEvent extends Event {
+	grid: DgridGrid;
+	row?: {};
+	cell?: {};
+	parentType?: string;
+}
+
+function buildSelectEvent(event: DgridSelectionEvent): SelectionData {
+	const selectionType = event.rows ? SelectionType.row : SelectionType.cell;
+	const data: SelectionData = {
+		type: selectionType,
+		data: []
+	};
+	if (selectionType === SelectionType.row) {
+		event.rows &&
+			(data.data = event.rows.map((rowData: any) => {
+				return {
+					item: rowData.data
+				};
+			}));
+	} else {
+		event.cells &&
+			(data.data = event.cells.map((cellData: any) => {
+				return {
+					item: cellData.row.data,
+					field: cellData.column.field
+				};
+			}));
+	}
+
+	return data;
+}
+
+function buildColumnStateChange(event: DgridColumnHiderEvent): ColumnStateChangeData {
+	return {
+		field: event.column.field,
+		id: event.column.id,
+		hidden: event.hidden
+	};
+}
+
+function buildColumnReorder(event: DgridColumnReorderEvent): ColumnReorderData {
+	return {
+		field: event.column.field,
+		id: event.column.id,
+		subRow: event.subRow
+	};
+}
+
+function buildColumnResize(event: DgridColumnResizeEvent): ColumnResizeData {
+	const column = event.grid.columns[event.columnId];
+	return {
+		field: column.field,
+		id: column.id,
+		width: event.width,
+		parentType: event.parentType
+	};
+}
+
+function buildSort(event: DgridSortEvent): SortData {
+	return {
+		parentType: event.parentType,
+		sort: event.sort
+	};
+}
+
+function buildError(event: DgridErrorEvent): ErrorData {
+	return {
+		error: event.error
+	};
+}
+
+function buildCellFocus(event: DgridCellFocusEvent): CellFocusData {
+	return {
+		row: event.row,
+		cell: event.cell,
+		parentType: event.parentType
+	};
+}
+
+function duplicateColumnDef(columnsSpec: Grid.ColumnSpec): Grid.ColumnSpec {
+	if (Array.isArray(columnsSpec)) {
+		if (columnsSpec.length === 0) {
+			return [];
+		}
+		return columnsSpec.map((columnSpec) => {
+			return duplicate(columnSpec);
+		});
+	} else {
+		return duplicate(columnsSpec);
+	}
+}
+
+function duplicateColumnSets(columnSets: Array<Array<Grid.Column[]>>): Array<Array<Grid.Column[]>> {
+	return columnSets.map((subRows) => {
+		return subRows.map((subRow) => {
+			return subRow.map((column) => duplicate(column));
+		});
+	});
+}
+
+function buildConstructor(properties: DgridInnerWrapperProperties, emitGridState: () => void): any {
+	const {
+		pagination = false,
+		keyboard = false,
+		selection = undefined,
+		selector = false,
+		tree = false,
+		columnHider = false,
+		columnReorder = false,
+		columnResizer = false,
+		compoundColumns = false,
+		columnSet = false
+	} = properties.features || {
```

Comment:
Since you're providing the defaults in lines 377-386, there shouldn't be a reason to give defaults in lines 388-397. This line can just be `} = properties.features || {};`